components/tikv_util/src/metrics/mod.rs

```diff
@@ -59,7 +116,20 @@ pub fn dump_to(w: &mut impl Write, should_simplify: bool) {
             MetricType::COUNTER => {
                 metrics.retain(|m| m.get_counter().get_value() > 0.0);
             }
-            MetricType::HISTOGRAM => metrics.retain(|m| m.get_histogram().get_sample_count() > 0),
+            MetricType::HISTOGRAM => {
+                let threshold = if simplify_level == MetricsCompactLevel::LoseLess {
```

Comment:
The simplify_level is always `NoCompact` since L105 has checked.

Reply:
if the level is `NoCompact` it will return at L109.